File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Python/s580342216_commented.py
import sys

class Solution:
    def __init__(self):
        for line in sys.stdin:
            self.run_case(line.strip())
    
    def run_case(self, line):
        parts = line.split()
        N = int(parts[0])
        K = int(parts[1])
        length = 0
        while N > 0:
            N = N // K
            length += 1
        print(length)

class LCS:
    def __init__(self):
        self.dp = []
    
    def lcs(self, A, B):
        m = len(A)
        n = len(B)
        self.dp = [[0]*(n+1) for _ in range(m+1)]
        for i in range(1, m+1):
            for j in range(1, n+1):
                if A[i-1] == B[j-1]:
                    self.dp[i][j] = self.dp[i-1][j-1] + 1
                else:
                    self.dp[i][j] = max(self.dp[i-1][j], self.dp[i][j-1])
        return self.dp[m][n]
    
    def getLCS(self, X, Y):
        m, n = len(X), len(Y)
        i, j = m, n
        lcs_list = []
        while i > 0 and j > 0:
            if X[i-1] == Y[j-1]:
                lcs_list.append(X[i-1])
                i -= 1
                j -= 1
            elif self.dp[i-1][j] > self.dp[i][j-1]:
                i -= 1
            else:
                j -= 1
        lcs_list.reverse()
        return ''.join(lcs_list)

if __name__ == "__main__":
    Solution()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s580342216_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Solution struct {
	scanner *bufio.Scanner
}

func NewSolution() *Solution {
	return &Solution{
		scanner: bufio.NewScanner(os.Stdin),
	}
}

func (s *Solution) Run() {
	for s.scanner.Scan() {
		s.runCase()
	}
}

func (s *Solution) runCase() {
	line := s.scanner.Text()
	parts := strings.Split(line, " ")
	N, err := strconv.Atoi(parts[0])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error parsing N: %v\n", err)
		return
	}
	K, err := strconv.Atoi(parts[1])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error parsing K: %v\n", err)
		return
	}

	len := 0
	for N > 0 {
		N /= K
		len++
	}

	fmt.Println(len)
}

func (s *Solution) strToIntArray(str string) []int {
	parts := strings.Split(str, " ")
	res := make([]int, len(parts))
	for i, part := range parts {
		res[i], _ = strconv.Atoi(part)
	}
	return res
}

type LCS struct {
	dp [][]int
}

func (l *LCS) LCS(A string, B string) int {
	szA := len(A)
	szB := len(B)
	dp := make([][]int, szA+1)
	for i := range dp {
		dp[i] = make([]int, szB+1)
	}

	for i := 0; i <= szA; i++ {
		dp[i][0] = 0
	}
	for j := 0; j <= szB; j++ {
		dp[0][j] = 0
	}

	for i := 1; i <= szA; i++ {
		for j := 1; j <= szB; j++ {
			if A[i-1] == B[j-1] {
				dp[i][j] = dp[i-1][j-1] + 1
			} else {
				dp[i][j] = max(dp[i-1][j], dp[i][j-1])
			}
		}
	}

	l.dp = dp
	return dp[szA][szB]
}

func (l *LCS) GetLCS(X string, Y string) string {
	m := len(X)
	n := len(Y)
	index := l.dp[m][n]
	temp := index

	lcs := make([]byte, index+1)
	lcs[index] = 0

	i := m
	j := n
	for i > 0 && j > 0 {
		if X[i-1] == Y[j-1] {
			lcs[index-1] = X[i-1]
			i--
			j--
			index--
		} else if l.dp[i-1][j] > l.dp[i][j-1] {
			i--
		} else {
			j--
		}
	}

	return string(lcs[:temp+1])
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	solution := NewSolution()
	solution.Run()
}
// End of Code
